core/datatests/tests/suites/optimising_line_formatter.rs

```diff
@@ -8,7 +8,25 @@ use crate::suites::get_input_output;
 use super::ErrorString;
 use pasfmt_core::prelude::*;
 
+struct ErrorLogger;
+impl log::Log for ErrorLogger {
+    fn enabled(&self, _metadata: &log::Metadata) -> bool {
+        true
+    }
+
+    fn log(&self, record: &log::Record) {
+        if matches!(record.level(), log::Level::Error) {
+            panic!("{:?} logged an error:\n{}", record.file(), record.args());
+        }
+    }
+
+    fn flush(&self) {}
+}
+static LOGGER: ErrorLogger = ErrorLogger {};
+
 fn run_test(input: &str) -> datatest_stable::Result<()> {
+    let _ = log::set_logger(&LOGGER).map(|()| log::set_max_level(log::LevelFilter::Trace));
```

Comment:
Fail if the logger is already set.
```suggestion
    let _ = log::set_logger(&LOGGER).expect("logger should not already be configured").map(|()| log::set_max_level(log::LevelFilter::Trace));
```

Reply:
While that makes sense on paper, the logger is not reset for each test run, and so that will fail every test except the first one. There is another solution, I am all ears.

Reply:
```rust
static SET_LOGGER: LazyLock<()> = LazyLock::new(|| {
    log::set_logger(&LOGGER).expect("logger should not already be configured");
    log::set_max_level(log::LevelFilter::Trace);
});

fn run_test(input: &str) -> datatest_stable::Result<()> {
    *SET_LOGGER;
```